src/compat/compat.h

```diff
@@ -34,19 +37,20 @@
 #include <unistd.h>
 #endif
 
+// We map Linux / BSD error functions and codes, to the equivalent
+// Windows definitions, and use the WSA* names throughout our code.
+// Note that glibc defines EWOULDBLOCK as EAGAIN (see errno.h).
 #ifndef WIN32
 typedef unsigned int SOCKET;
-#include <errno.h>
+#include <cerrno>
 #define WSAGetLastError()   errno
 #define WSAEINVAL           EINVAL
-#define WSAEALREADY         EALREADY
 #define WSAEWOULDBLOCK      EWOULDBLOCK
 #define WSAEAGAIN           EAGAIN
 #define WSAEMSGSIZE         EMSGSIZE
 #define WSAEINTR            EINTR
 #define WSAEINPROGRESS      EINPROGRESS
 #define WSAEADDRINUSE       EADDRINUSE
-#define WSAENOTSOCK         EBADF
 #define INVALID_SOCKET      (SOCKET)(~0)
 #define SOCKET_ERROR        -1
 #else
```

Comment:
@vasild could you explain what the following block
```cpp
#ifndef WSAEAGAIN
#ifdef EAGAIN
#define WSAEAGAIN EAGAIN
#else
#define WSAEAGAIN WSAEWOULDBLOCK
#endif
#endif
```
is accounting for (given you added it)? I'd have thought everything inside this #else block could be replaced by `#define WSAEAGAIN WSAEWOULDBLOCK`, as `WSAEAGAIN` would never be defined?

Reply:
The purpose of that is to ensure that `WSAEAGAIN` is defined on Windows - if it is already defined, don't do anything. If it is not defined but `EAGAIN` is defined then define `WSAEAGAIN` to `EAGAIN`, otherwise define `WSAEAGAIN` to `WSAEWOULDBLOCK` which is assumed to be present.

Reply:
Right, but when would that be the case? I find the code confusing because I'm pretty sure it's trying to account for situations that don't exist. Can you point to a toolchain / system where we'd be compiling Bitcoin Core, for Windows, where `WSAEAGAIN` would be defined? (so we can document it)?

>  to WSAEWOULDBLOCK which is assumed to be present.

If you're going to assume `WSAEWOULDBLOCK` is present, I think we can assume `WSAEAGAIN` wont be, and again, just simplify to `#define WSAEAGAIN WSAEWOULDBLOCK`.

Reply:
> Can you point to a toolchain...

No, I just coded it defensively so that it works in all scenarios, including future platforms that may define `WSAEAGAIN` or aux headers from dependencies - e.g. leveldb, sqlite may define it even if Windows itself does not.

But ok, if you feel like simplifying it, then I am fine with that. Worst that can happen is that it will fail to compile with an error like "`WSAEAGAIN` is redefined" - no subtle behavior at runtime. So, it would become:

```cpp
#ifdef EAGAIN
#define WSAEAGAIN EAGAIN
#else
#define WSAEAGAIN WSAEWOULDBLOCK
#endif
```

Reply:
> Worst that can happen is that it will fail to compile with an error like "WSAEAGAIN is redefined"

I think that would be much better behaviour, and make it obvious if such a situation where to exist, rather than silently accomodating dependencies / toolchains changing underneath us with some #ifdef soup.